### ISSUE TITLE
onnx-tensorrt does not support reshape layer with input 3 dims

### ISSUE THREAD

**Issue by feitiandemiaomi** [onnx/onnx-tensorrt#442]
```
def forward(self, x):
       out = self.features(x)
       out = self.avgpool(out)
       out = out.permute(3, 0, 1, 2).view(out.size(3), out.size(0), -1)
       out = self.classifier(out)
       out = F.log_softmax(out, dim=2)
       return out
```

**Here are my code and  all the operations on shapes:
[1, 3, 32, 200] ->[1, 384, 4, 25] ->[1, 384, 1, 25] ->[25, 1, 384] ->[25, 1, 5990]**
When I change my pytorch model  into trt , I meet the first problem:
like #87  **While parsing node number 26 [Transpose -> "96"]:**
and I change my code like this:
```
def forward(self, x):
    out = self.features(x)
    out = self.avgpool(out)
    #out = out.permute(0, 3, 2, 1).view(25, 1, 384)
    out = out.permute(0, 3, 2, 1).reshape(25, 1, 384)
    out = self.classifier(out)
    out = F.log_softmax(out, dim=2)
    return out
```
**the bug solved, but have a new problem :**
![图片](https://user-images.githubusercontent.com/18325462/80082640-39802500-8587-11ea-922f-992472844278.png)

I notice that the code in builtin_op_importers.cpp:1530
![图片](https://user-images.githubusercontent.com/18325462/80083075-c7f4a680-8587-11ea-8c70-aef5facaded5.png) 
There has a remove_dim op,which removes the dim 25 and causes an error, so I do not know why this happen ,Is it because onnx-tensorrt does not support reshape layer or view layer with input 3 dims?
@onnx @yinghai @perrywu1989 @mgolub2 @thefacetakt

Hope to get some suggestions,Thanks a lot. 
PS: The code can inference rightly

**Comment by feitiandemiaomi** [comment]
Tensorrt version : 5.1.5

**Comment by thefacetakt** [comment]
Hi. As i mentioned in https://github.com/onnx/onnx-tensorrt/issues/66, `apparently, TRT has some issues with reshapes, but nothing against squeeze/unsquese`.
I think your `out = out.permute(0, 3, 2, 1).reshape(25, 1, 384)` can be replaced with `out = out.permute(0, 3, 2, 1).squeeze(0)`. I guess, it should help with "get_shape_size" assertion. 

However, I guess your `self.classifier(out)` may have some FC inside. In TF code this lead to combination of ops which I wasn't able to convert to trt successfully. Don't know if it is the case for pytorch. Anyway, I was able to overcome it replacing FC with 1x1 conv (which I also pointed out in https://github.com/onnx/onnx-tensorrt/issues/66)

Hope it helps!

**Comment by feitiandemiaomi** [comment]
@thefacetakt You are right , `out = out.permute(0, 3, 2, 1).squeeze(0)`  indeed can solve "get_shape_size" ,but There has an another problem :

![图片](https://user-images.githubusercontent.com/18325462/80092920-2c1e6700-8596-11ea-8fad-98c09488d533.png)
And I try to change into 
`out = out.permute(0, 3, 2, 1).reshape(25,384)
 out = out.unsqueeze(1)` 

There has the same "get_shape_size" error ,so onnx-tensorrt does not support reshape layer with input 2 dims ,it just support one dim.

**Comment by kevinch-nv** [comment]
@feitiandemiaomi we've made some improvements in later TRT versions to properly handle these reshape cases. Can you upgrade to the latest TRT version and try importing your model again?

**Comment by kevinch-nv** [comment]
Closing due to no activity in this thread, if you are still having troubles feel free to open a new issue with a repro on the latest version of TRT.

**Comment by Semen52** [comment]
> @feitiandemiaomi we've made some improvements in later TRT versions to properly handle these reshape cases. Can you upgrade to the latest TRT version and try importing your model again?

Hi,
Could you please provide a version where fix was applied?

**Comment by kevinch-nv** [comment]
@Semen52 please try the master branch as it contains the most up-to-date changes.

If you are still seeing an error, please open a new issue.
